idc_index/index.py

```diff
@@ -270,15 +276,67 @@ def download_from_selection(self, downloadDir=None, dry_run=True, collection_id=
 
     Raises:
     """
-    def download_from_manifest(self, manifest_file, downloadDir):
-        cmd = [self.s5cmdPath, '--no-sign-request', '--endpoint-url', 'https://s3.amazonaws.com', 'run',
-            manifest_file, downloadDir]
-        process = subprocess.run(cmd, capture_output=True, text=True)
-        logging.info(process.stderr)
+    def download_from_manifest(self, manifestFile, downloadDir, quiet=True):
+
+        downloadDir = os.path.abspath(downloadDir)
+
+        if not os.path.exists(downloadDir):
+            raise ValueError("Download directory does not exist.")
+        if not os.path.exists(manifestFile):
+            raise ValueError("Manifest does not exist.")
+
+        # open manifest_file and read the first line that does not start from '#'
+        with open(manifestFile, 'r') as f:
+            for line in f:
+                if not line.startswith('#'):
+                    break
+        pattern = r"(s3:\/\/.*)\/\*"
+        match = re.search(pattern, line)
+        if match is None:
+            logger.error("Could not find the bucket URL in the first line of the manifest file.")
+            return
+        folder_url = match.group(1)
+     
+        cmd = [self.s5cmdPath, '--no-sign-request', '--endpoint-url', self.aws_endpoint_url, 'ls', folder_url]
+        process = subprocess.run(cmd, capture_output=(not quiet), text=(not quiet))
+        # check if output starts with ERROR
+        if process.stderr.startswith('ERROR'):
```

Comment:
![image](https://github.com/ImagingDataCommons/idc-index/assets/115020590/f77601d2-4e22-4c78-a238-c85ab21a71bf)


![image](https://github.com/ImagingDataCommons/idc-index/assets/115020590/93a37050-726e-4438-a836-c39066b39a09)

Reply:
I'll try this on a jetstream vm too, just in case. And I realized I cannot run the tests on windows yet because of the prerequisites.

Reply:
I could replicate the same errors on a jetstream vm too